### ISSUE TITLE
panic: counter cannot decrease in value

### ISSUE THREAD

**Issue by jonfunkhouser** [DataDog/datadog-agent#10730]
datadog-agent 7.33.0 is terminating shortly after startup with the following error:

```
panic: counter cannot decrease in value

goroutine 650 [running]:
github.com/prometheus/client_golang/prometheus.(*counter).Add(0xc000d72600, 0xc12ebe0c00000000)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/counter.go:109 +0x125
github.com/DataDog/datadog-agent/pkg/telemetry.(*promCounter).Add(0xc000186938, 0xc12ebe0c00000000, 0x0, 0x0, 0x0)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/telemetry/prom_counter.go:29 +0x6c
github.com/DataDog/datadog-agent/pkg/serializer/stream.(*JSONPayloadBuilder).BuildWithOnErrItemTooBigPolicy(0xc000af25d0, 0x5174b28, 0xc000e9b050, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/serializer/stream/json_payload_builder.go:119 +0x168
github.com/DataDog/datadog-agent/pkg/serializer.Serializer.serializeStreamablePayload(...)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/serializer/serializer.go:204
github.com/DataDog/datadog-agent/pkg/serializer.(*Serializer).SendServiceChecks(0xc0006db280, 0x5174b28, 0xc000e9b050, 0xc000e9b050, 0x1)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/serializer/serializer.go:285 +0xb3
github.com/DataDog/datadog-agent/pkg/aggregator.(*BufferedAggregator).sendServiceChecks(0xc000adf080, 0xc07649712745e040, 0x18d15f15ce, 0x6e1ffc0, 0xc000839180, 0xe, 0x10)
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/aggregator.go:612 +0x11b
created by github.com/DataDog/datadog-agent/pkg/aggregator.(*BufferedAggregator).flushServiceChecks
	/omnibus/src/datadog-agent/src/github.com/DataDog/datadog-agent/pkg/aggregator/aggregator.go:645 +0x47a
```

**Describe what happened:**
I have datadog-agent set up in a Docker container to profile a dotnet 6 application running in another container. `datadog-agent` is crashing some time after its started.

**Describe what you expected:**
I expected `datadog-agent` to continue running.

**Additional environment details (Operating System, Cloud provider, etc):**
Cloud Provider: EC2, t3.medium instance
Operating System: Ubuntu 20.04.3 LTS
Docker: 20.10.12, build e91ed57
datadog-agent: 7.33.0
datadog-agent Docker image: `gcr.io/datadoghq/agent                                               7               df69f3f11162   6 days ago     857MB`
dotnet: 6.0.1
dd-trace-dotnet: 2.1.0

Volumes mounted on container:
```
/var/run/docker.sock:/var/run/docker.sock:ro
/proc/:/host/proc/:ro
/sys/fs/cgroup/:/host/sys/fs/cgroup:ro
/etc/os-release:/host/etc/os-release:ro
 ```

Environment Variables passed to datadog-agent:
```
DD_LOG_LEVEL=debug
DD_APM_ENABLED=true
DD_APM_NON_LOCAL_TRAFFIC=true
DD_DOGSTATSD_NON_LOCAL_TRAFFIC=true
DD_API_KEY=******************************
```

Log file from agent is attached:
[datadog-agent.log](https://github.com/DataDog/datadog-agent/files/7975756/datadog-agent.log)

**Comment by sgnn7** [comment]
Hi @jonfunkhouser,
While Golang is supposed to have guards in the language itself against that, it seems like that your particular combination of hardware/OS seems to report monotonic time incorrectly in some circumstances. I created an internal task to guard against this situation but it may take a while for a release with a fix to roll out. As a workaround, you could maybe try setting `enable_json_stream_shared_compressor_buffers: false` in your configuration to see if avoiding this block will get your agent to not crash until that time (at a cost of some performance in sending throughput).

We can leave the issue open for now on this until a fix lands.

**Comment by djmitche** [comment]
This should be fixed in 7.35.0.

**Comment by rfroshaug** [comment]
I'm also seeing this issue in the latest datadog-cluster-agent, will that also contain the fix?
